DelvEdit/src/com/interrupt/dungeoneer/editor/EditorCameraController.java

```diff
@@ -63,124 +66,132 @@ public void tick() {
+
+        ya *= 0.8;
+
+        float xm = 0f;
+        float zm = 0f;
+
+        if (input.isKeyPressed(Input.Keys.A)) {
+            xm = -1f;
+        }
+        if (input.isKeyPressed(Input.Keys.D)) {
+            xm = 1f;
+        }
+
+        if (input.isKeyPressed(Input.Keys.W) || scrollAmount < 0) {
+            zm = -1f;
+        }
+        if (input.isKeyPressed(Input.Keys.S) || scrollAmount > 0) {
+            zm = 1f;
+        }
+
+        if (scrollAmount < 0) {
+            za -= scrollSpeed;
+        } else if (scrollAmount > 0) {
+            za += scrollSpeed;
+        }
+
+        zm += za;
+
+        za *= 0.8f;
+
+        if (input.isKeyPressed(Input.Keys.Q) && !input.isKeyPressed(Input.Keys.SHIFT_LEFT)) {
+            position.z -= 0.1f;
+        }
+        if (input.isKeyPressed(Input.Keys.E) && !input.isKeyPressed(Input.Keys.SHIFT_LEFT)) {
+            position.z += 0.1f;
+        }
+
+        if (input.isKeyPressed(Input.Keys.SHIFT_LEFT)) {
+            xm *= 2.0f;
+            zm *= 2.0f;
+        }
+
+        orbitDistance += zm * walkSpeed;
+
+        position.z += (zm * Math.sin(rotation.y)) * walkSpeed;
+        zm *= Math.cos(rotation.y);
+        position.x += (xm * Math.cos(rotation.x) + zm * Math.sin(rotation.x)) * walkSpeed;
+        position.y += (zm * Math.cos(rotation.x) - xm * Math.sin(rotation.x)) * walkSpeed;
+
+        Player player = Editor.app.player;
+        if (player != null) {
+            player.rot = rotation.x;
+            player.yrot = rotation.y;
+
+            player.xa += (xm * Math.cos(rotation.x) + zm * Math.sin(rotation.x)) * 0.025f * Math.min(player.friction * 1.4f, 1f);
+            player.ya += (zm * Math.cos(rotation.x) - xm * Math.sin(rotation.x)) * 0.025f * Math.min(player.friction * 1.4f, 1f);
+        }
+
+        if (Gdx.input.isKeyPressed(Input.Keys.ALT_LEFT) && (input.isButtonPressed(Input.Buttons.RIGHT) || turnLeft || turnRight || turnUp || turnDown)) {
+            // Calculate the next camera direction vector;
+            Vector3 cameraNewDirection = new Vector3(0, 0, 1);
+            cameraNewDirection.rotate(rotation.y * 57.2957795f, 1f, 0, 0);
+            cameraNewDirection.rotate((float) (rotation.x + 3.14) * 57.2957795f, 0, 1f, 0);
+            cameraNewDirection.nor();
+
+            // Calculate the orbit pivot.
+            if (orbitDistance < 0) {
+                orbitDistance = 3.0f;
+            }
+
+            Vector3 pivotPosition = new Vector3(camera.direction).scl(orbitDistance).add(camera.position);
+
+            // Calculate new camera position.
+            cameraNewDirection.scl(-orbitDistance);
+            cameraNewDirection.add(pivotPosition);
+
+            position.set(cameraNewDirection.x, cameraNewDirection.z, cameraNewDirection.y);
+        }
+
+        if (Gdx.input.isKeyPressed(Input.Keys.SHIFT_LEFT) && Gdx.input.isButtonPressed(Input.Buttons.RIGHT)) { // Pan mode enabled (SHIFT_LEFT held)
```

Comment:
Whenever I see boolean logic with a little note next to it, my instinct is to wrap that up in a method. Maybe something like:
```java
if (isPanning()) { ... }
```

Reply:
I intended to unwrap these while refactoring the controller code and making these checks/states more clear, actually. I can amend this with my clean-up if that's better?